src/cryptoadvance/specter/internal_node.py

```diff
@@ -76,6 +76,8 @@ def __init__(
                 self.datadir = os.path.join(self.datadir, "signet")
             logger.info(f"persisting {self} in __init__")
             write_node(self, self.fullpath)
+        if not os.path.isdir(os.path.expanduser(self.datadir)):
+            os.makedirs(os.path.expanduser(self.datadir))
```

Comment:
I just recently replaced this idiom in device_manager because of some strange race-condition where the error-message claimed that the directory already exists. Can you replace it with this:
```suggestion
        pathlib.Path(self.datadir).mkdir(parents=True, exist_ok=True)
```